spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/query/derived/DerivedGraphRepositoryQuery.java

```diff
@@ -147,21 +156,57 @@ private Filters resolveParams(Object[] parameters) {
 		Map<Integer, Object> params = new HashMap<>();
 
 		for (int i = 0; i < parameters.length; i++) {
-			if (graphQueryMethod.getQueryDepthParamIndex() == null || (graphQueryMethod.getQueryDepthParamIndex() != null && graphQueryMethod.getQueryDepthParamIndex() != i)) {
+			if (graphQueryMethod.getQueryDepthParamIndex() == null
+					|| (graphQueryMethod.getQueryDepthParamIndex() != null && graphQueryMethod.getQueryDepthParamIndex() != i)) {
 				params.put(i, parameters[i]);
 			}
 		}
 		List<CypherFilter> cypherFilters = queryDefinition.getCypherFilters();
 		Filters queryParams = new Filters();
 		for (CypherFilter cypherFilter : cypherFilters) {
 			Filter filter = cypherFilter.toFilter();
-			filter.setPropertyValue(params.get(cypherFilter.getPropertyPosition()));
+
+			FilterFunction function = filter.getFunction();
+			Object functionValue = function instanceof DistanceComparison ?
+					extractDistanceArgs(params, cypherFilter.getPropertyPosition()) :
+					params.get(cypherFilter.getPropertyPosition());
+			function.setValue(functionValue);
 			queryParams.add(filter);
 		}
 		return queryParams;
 	}
 
 
+	private DistanceFromPoint extractDistanceArgs(Map<Integer, Object> params, int startIndex) {
+		Object firstArg = params.get(startIndex);
```

Comment:
instanceof is in fact null safe. So we'll get a meaningful exception in any case where we don't have exactly one Point and one Distance, in any order.